chapter16/qli/ex4.pl

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/perl
+use warnings;
+use strict;
+
+my $counter = {};
+print "signal handler started with pid[$$] \n";
+
+#Overwriting By values is the most efficient compare to 
+#@SIG{ keys %SIG } = ( 'sig_handler' ) x keys %SIG;
+#$SIG{$_} = 'sig_handler' foreach keys %SIG;
+$_ = 'sig_handler' foreach values %SIG;
```

Comment:
While $_ is a alias to the current loop variable, I'd usually advise against modifying it in this way to avoid unexpected outcomes. Looping over the keys of %SIG then updating $SIG{$_} isn't any less efficient and conceptually tidier.